src/tuple_impl.rs

```diff
@@ -57,12 +57,12 @@ impl<T> Iterator for TupleBuffer<T>
 
     fn size_hint(&self) -> (usize, Option<usize>) {
         let buffer = &self.buf.as_ref()[self.cur..];
-        let len = if buffer.len() == 0 {
+        let len = if buffer.is_empty() {
             0
         } else {
             buffer.iter()
                   .position(|x| x.is_none())
-                  .unwrap_or(buffer.len())
+                  .unwrap_or_else(|| buffer.len())
```

Comment:
Calling`.len` is *really* cheap. Adding the indirection of a thunk will make more work for the compiler, with probably little (to no) gain in performance.

Reply:
Fair point. When encountering such situations in own projects, I'm inclined to go with clippy, because when this warning does not show up anymore, I can focus more easily on real issues in subsequent clippy runs.

If you still prefer that I should remove it, please let confirm this, and I'll revert it.

Reply:
> I'm inclined to go with clippy, because when this warning does not show up anymore, I can focus more easily on real issues in subsequent clippy runs

I think if this change is reverted, the surrounding functions should have `#[allow(clippy::lint_name)]`.

Reply:
I'm fine with `unwrap_or_else`, I think. This `len()` is coming from `ExactSizeIterator`, so the call isn't *necessarily* cheap.